testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/AssertWithRetryBuilder.kt

```diff
@@ -9,14 +9,10 @@ class AssertWithRetryArgs {
     var timeout: Duration = Duration.ofSeconds(10)
     var interval: Duration = Duration.ofMillis(500)
     var startDelay: Duration = Duration.ofMillis(10)
-    var command: (() -> SimpleResponse)? = null
+    var command: (() -> SimpleResponse) = { throw IllegalArgumentException("Command not specified in AssertWithRetryArgs") }
```

Comment:
Would it be best not to default it at all, that way there will be a compile-time error if people "forget" to supply command. Rather than having a runtime error.

Reply:
This is a small class and the definite assignment check acts within it. So not setting the default triggers a compile error in this class, before we get to considering its users.

Having said that, it is possible to build what you suggest using the fluent builder pattern that's popular in C#. The idea there is the functions to set properties return different interface types, guiding the API user towards completing the data the object needs. The API user only gets the final "you can use the object" type once they have completed what they need. In this case, we would return the type that allows use when they set command.

Reply:
Makes sense how about:
```suggestion
    lateinit var command: (() -> SimpleResponse)
```
?

Reply:
Done